roles/ui/files/FWO_UI/Pages/Certification.razor

```diff
@@ -12,33 +12,51 @@
 
 @* ==== LEFT SIDEBAR ==== *@
 <Sidebar @ref="deviceSelectionSidebar" Collapsible="true" Resizeable="true" PositionLeft="true" @bind-Width="sidebarLeftWidth">
-    <DeviceSelection Managements="managements" @bind-FilterInput="filterInput" @bind-SelectAll="selectAll" ></DeviceSelection>
-</Sidebar>
-
-<div style="margin-left: @($"{sidebarLeftWidth + 10}px"); margin-right: @($"{sidebarRightWidth + 10}px");">
-    <h3>@(userConfig.GetText("recertification"))</h3>
+    
     <div class="d-flex flex-row flex-wrap mb-2">
         <div class="mr-3">
             <div class="input-group input-group-sm">
-                <div class="input-group-prepend">
-                    <label class="input-group-text" for="inputGroupSelect01">@(userConfig.GetText("due_within")):</label>
-                </div>
-                <input id="inputGroupSelect01" type="text" class="form-control form-control-sm" @bind="recertificationDisplayPeriod" />
-                <div class="input-group-append">
-                    <button class="btn btn-sm btn-primary" @onclick="() => GenerateRecertificationReport()">@(userConfig.GetText("load_rules"))</button>
-                </div>
+                <table style="margin-left: 10px; margin-top: 10px;">
+                    <tr>
+                        <td>
+                            <div class="input-group-prepend">
+                                <label for="inputGroupSelect01">@(userConfig.GetText("due_within")):</label>
+                            </div>
+                        </td>
+                        <td>
+                            <input id="inputGroupSelect01" type="text" class="form-control form-control-sm" @bind="recertificationDisplayPeriod" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>
+                            <div class="input-group-append">
+                                <button class="btn btn-sm btn-primary" @onclick="() => GenerateRecertificationReport()">@(userConfig.GetText("load_rules"))</button>
+                            </div>
+                        </td>
+                        <td>
+                            <AuthorizeView Roles="admin, recertifier">
+                                <Authorized>
+                                    <button class="btn btn-sm btn-primary" @onclick="() => RequestExecuteSelected()">@(userConfig.GetText("execute_selected"))</button>
```

Comment:
Yes it`s logical to have the rule selection etc in the left sidebar. Maybe we'll find a way to make it usable a bit more intuitively. In a later step I would try to place the execute button back to somewhere in the middle part, if we find a solution not to lose it when scrolling.

Reply:
Agree